File: SAV_Backend/SAV_Backend/Models/NotificationClient.cs
using System.Text.Json.Serialization;

namespace SAV_Backend.Models
{
    public class NotificationClient
    {
        public int NotificationClientId { get; set; }
        public int SenderId { get; set; }
        public string SenderName { get; set; }
        public int ReceiverId { get; set; }

        public string Message { get; set; }
        public bool IsRead { get; set; } = false;
        public bool IsVisited { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        [JsonIgnore]
        public virtual Client? client { get; set; }

    }
}

File: SAV_Frontend/src/Models/NotificationClient.ts
import { Client } from "./Client";

export interface NotificationClient {
    notificationClientId: number;
    senderId: number;
    senderName: string;
    receiverId: number;
    message: string;
    isRead: boolean;
    isVisited: boolean;
    createdAt: Date;
    client?: Client; // Remplacez 'Client' par l'interface correspondante si elle existe
  }
  